reyna/src/com/b2msolutions/reyna/system/Preferences.java

```diff
@@ -196,4 +199,25 @@ private String getString(String key, String defaultValue) {
         SharedPreferences sp = this.context.getSharedPreferences(Preferences.class.getName(), Context.MODE_PRIVATE);
         return sp.getString(key, defaultValue);
     }
+
+    public long getNonRecurringWwanBlackoutStartTime() {
+        return this.getLong(WWAN_BLACKOUT_START, -1);
```

Comment:
Probably not relevant in practise (in particular as both start and end are set to -1 at the same time, so not even -1 satisfy inBetween condition), but theoretical -1 is a valid epoch time stamp, so not best for indicating missing value.